pkg/routes/routes.go

```diff
@@ -21,14 +21,15 @@ type DMSRouter struct {
 // Pingdom is the endpoint that pingdom will query, it returns 200 if all the checks are within the heartbearExpiration
 // and it returns 500 if any of the checks are outside of the heartbearExpiration
 func (r *DMSRouter) Pingdom(c echo.Context) error {
-	all, err := r.Store.All()
+	all, err := r.Store.AllCheckins()
 	if err != nil {
 		logrus.WithError(err).Error("unable to retrieve all stored data from etcd")
 	}
 	for k, v := range all {
-		if v.Last != (time.Time{}) {
-			if time.Since(v.Last) > (r.HeartbeatExpiration) {
+		if v != (time.Time{}) {
+			if time.Since(v) > (r.HeartbeatExpiration) {
```

Comment:
time.Since might return invalid values when time value is zero, but has a timezone.

Similarly, usually it's nice to handle all of the loop elements with the exact same time interval. Currently you are calling time.Now multiple times implicitly (time.Since) and explicitly.

Writing `now := time.Now()` in the beginning of the func and using that throughout the func avoids some issues.